Bottom: 2daa33cdc5b40d2b86170f7a5d5b1be6880e2af2
Top:    50e42976806f62ae588ecf5700f4c9ac7e995948
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-06 16:12:17 +0000

Workaround crash where contour is passed as NULL to poly_PreContour














---

diff --git a/src/polygon.c b/src/polygon.c
index 181ba6e..fd68900 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -235,6 +235,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
